src/js/index.js

```diff
@@ -1 +1,47 @@
 import "../styles/main.scss";
+
+const $ = (selector) => document.querySelector(selector);
+
+const emailRegex = /^(([^<>()[\]\\.,;:\s@\"]+(\.[^<>()[\]\\.,;:\s@\"]+)*)|(\".+\"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/;
+
+$(".js-email-input").addEventListener("keyup", (e) => {
+    const email = e.target.value;
+    if (!emailRegex.test(email)) {
+        $(".js-error").textContent = "Please enter a valid email";
+        $(".js-submit-button").disabled = true;
+    }
+    else {
+        $(".js-error").textContent = "";
+        $(".js-submit-button").disabled = false;
+    }
+})
+$(".js-form").addEventListener("submit", (e) => e.preventDefault());
+
+$(".js-submit-button").addEventListener("click", () => {
+    const email = $(".js-email-input").value;
+    if (!emailRegex.test(email)) {
+        $(".js-error").textContent = "Please enter a valid email";
+        $(".js-submit-button").disabled = true;
+    }
+    else {
+        $(".js-error").textContent = "";
+        $(".js-submit-button").disabled = false;
+        let httpRequest;
+        if (window.XMLHttpRequest) {
+            httpRequest = new XMLHttpRequest();
+        }
+        else if (window.ActiveXObject) {
+            httpRequest = new ActiveXObject("Microsoft.XMLHTTP");
```

Comment:
Is ActiveXObject still needed in recent M$ browsers?

Reply:
`ActiveXObject` is supported in IE only. See https://developer.mozilla.org/en-US/docs/Web/JavaScript/Microsoft_Extensions/ActiveXObject

Reply:
> 
> 
> `ActiveXObject` is supported in IE only. See https://developer.mozilla.org/en-US/docs/Web/JavaScript/Microsoft_Extensions/ActiveXObject . It's not needed in modern browsers

So the question is: **Do we care?** IE is somewhere in the 1 - 4 % range, depending on how you measure it. And less if you consider that many use IE because they have to for a specific application and use another browser the rest of the time.

Reply:
I looked it up, and IE seems to have ~ 1.7% coverage. However, I'm for keeping this, as it has a relatively small cost only AFAIK and the benefit for those users who want to or must use IE is relatively large.